### PR TITLE
Added RestLayer Changes for PIT stats

### PR DESCRIPTION
**Pull request by ajaymovva**
Signed-off-by: Ajay Kumar Movva <movvaam@amazon.com>

### Description
Added Rest Layer Changes for PIT stats.

Service Layer Changes: [#4030 ](https://github.com/opensearch-project/OpenSearch/pull/4030)
 
### Issues Resolved
[#1147](https://github.com/opensearch-project/OpenSearch/issues/1147)

### Sample Output
1. Index Creation: `localhost:9200/my-index-000001/_doc/`.
2. Creation multiple PIT 's.
3. Node Stats API output PIT's

```
 "search" : {
          "open_contexts" : 0,
          "query_total" : 0,
          "query_time_in_millis" : 0,
          "query_current" : 0,
          "fetch_total" : 0,
          "fetch_time_in_millis" : 0,
          "fetch_current" : 0,
          "scroll_total" : 0,
          "scroll_time_in_millis" : 0,
          "scroll_current" : 0,
          "point_in_time_total" : 4,
          "point_in_time_time_in_millis" : 379478,
          "point_in_time_current" : 0,
          "suggest_total" : 0,
          "suggest_time_in_millis" : 0,
          "suggest_current" : 0
       },
```
Cat Indices Output:

```
❯ curl http://localhost:9200/_cat/indices\?v\&h\=search.point_in_time_current,search.point_in_time_time
search.point_in_time_current search.point_in_time_time
                           0                            6.3m
```

Cat Nodes Output:

```
❯ curl http://localhost:9200/_cat/nodes\?v\&h\=search.point_in_time_current,search.point_in_time_time
search.point_in_time_current search.point_in_time_time
                           0                            6.3m
```
Cat Shards Output:
```
❯ curl http://localhost:9200/_cat/shards\?v\&h\=search.point_in_time_current,search.point_in_time_time
search.point_in_time_current search.point_in_time_time
                           0                            6.3m
```

 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/rest/action/cat/RestNodesAction.java [hunk 1]
```diff
@@ -310,6 +310,19 @@ protected Table getTableWithHeader(final RestRequest request) {
         );
         table.addCell("search.scroll_total", "alias:scto,searchScrollTotal;default:false;text-align:right;desc:completed scroll contexts");
 
+        table.addCell(
+            "search.point_in_time_current",
+            "alias:scc,searchPointInTimeCurrent;default:false;text-align:right;desc:open pit contexts"
+        );
+        table.addCell(
+            "search.point_in_time_time",
+            "alias:scti,searchPointInTimeTime;default:false;text-align:right;desc:time pit contexts held open"
```
Comment (bharath-techie) [inline server/src/main/java/org/opensearch/rest/action/cat/RestNodesAction.java:319]: lets use term 'point in time' instead of 'pit' everywhere

Comment (ajaymovva) [reply server/src/main/java/org/opensearch/rest/action/cat/RestNodesAction.java]: Updated

**Review by bharath-techie — APPROVED** [review]

File: server/src/main/java/org/opensearch/rest/action/cat/RestShardsAction.java [hunk 2]
```diff
@@ -225,6 +225,18 @@ protected Table getTableWithHeader(final RestRequest request) {
...
         );
         table.addCell("search.scroll_total", "alias:scto,searchScrollTotal;default:false;text-align:right;desc:completed scroll contexts");
+        table.addCell(
+            "search.point_in_time_current",
+            "alias:spc,searchPointInTimeCurrent;default:false;text-align:right;desc:open point in time contexts"
+        );
+        table.addCell(
+            "search.point_in_time_time",
+            "alias:spti,searchPointInTimeTime;default:false;text-align:right;desc:time point in time contexts held open"
+        );
```
Comment (Bukhtawar) [inline server/src/main/java/org/opensearch/rest/action/cat/RestShardsAction.java:235]: nit: call it `search.point_in_time_total_time`

Comment (ajaymovva) [reply server/src/main/java/org/opensearch/rest/action/cat/RestShardsAction.java]: Updated

**Review by pranikum — APPROVED** [review]

**Comment by dhruv16dhr** [comment]
@reta @Bukhtawar Please review and approve this PR

**Review by reta — APPROVED** [review]

**Review by Bukhtawar — APPROVED** [review]

**Pull request merged**